spec/controllers/sessions_controller_spec.rb

```diff
@@ -27,130 +27,164 @@
 
-        it "creates a session" do
-          subject
-          expect(session[:user_id]).to eq(user.id)
-        end
+          it "creates a session" do
+            subject
+            expect(session[:user_id]).to eq(user.id)
+          end
 
-        it "redirects to the member root path" do
-          subject
-          expect(response).to redirect_to members_root_path
+          it "redirects to the application edit page" do
+            subject
+            expect(response).to redirect_to edit_application_path(user.application)
+          end
         end
 
-        it "creates session for member" do
-          expect { subject }.to_not change { User.count }
-          expect(session[:user_id]).to eq(user.id)
-        end
+        context "who is a member" do
+          before { user.update_attribute(:state, "member") }
 
-        context "who is already logged in" do
-          let(:auth) { create :authentication, user: user }
+          it "doesn't make a new user" do
+            expect { subject }.to_not change { User.count }
+          end
 
-          before do
-            login_as(user)
+          it "creates a session" do
+            subject
+            expect(session[:user_id]).to eq(user.id)
           end
 
           it "redirects to the member root path" do
             subject
             expect(response).to redirect_to members_root_path
           end
+
+          it "creates session for member" do
+            expect { subject }.to_not change { User.count }
+            expect(session[:user_id]).to eq(user.id)
+          end
+
+          context "who is already logged in" do
+            let(:auth) { create :authentication, user: user }
+
+            before do
+              login_as(user)
+            end
+
+            it "redirects to the member root path" do
+              subject
+              expect(response).to redirect_to members_root_path
+            end
+          end
         end
       end
-    end
 
-    describe "with an existing, logged-in user" do
-      let(:user) { create_with_omniauth(OmniAuth.config.mock_auth[:github]) }
+      describe "with an existing, logged-in user" do
+        let(:user) { create_with_omniauth(OmniAuth.config.mock_auth[:github]) }
+
+        subject { get :create, params: { provider: "google_oauth2" } }
 
-      subject { get :create, params: { provider: "google_oauth2" } }
+        before do
+          log_in(user)
+          request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:google_oauth2]
+        end
+
+        it "creates a new authentication for the user" do
+          expect { subject }.to change { user.authentications.count }.from(1).to(2)
+        end
+
+        it "makes a new Google authentication" do
+          subject
+          expect(user.authentications.last.provider).to eq "google_oauth2"
+        end
+
+        it "prints the thing again" do
+          subject
+        end
+      end
+    end
 
+    describe "with Google authentication" do
```

Comment:
This `describe "with Google authentication"` block (lines 162-187) is completely new -- it's confusing that Github diffs it against other test cases that were already there.